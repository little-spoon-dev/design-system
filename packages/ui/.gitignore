# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.

# build
*.js
*.js.flow
*.js.map
*.d.ts
*.d.ts.map
