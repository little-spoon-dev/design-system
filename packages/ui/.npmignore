# See https://npm.github.io/publishing-pkgs-docs/publishing/the-npmignore-file.html for more about ignoring files.

*.json
coverage
jest.config.ts
__tests__
