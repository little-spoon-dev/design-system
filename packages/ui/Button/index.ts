export { default } from '@littlespoon/button'
export * from '@littlespoon/button'
