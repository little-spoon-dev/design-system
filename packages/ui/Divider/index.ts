export { default } from '@littlespoon/divider'
export * from '@littlespoon/divider'
