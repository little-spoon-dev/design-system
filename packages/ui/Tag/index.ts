export { default } from '@littlespoon/tag'
export * from '@littlespoon/tag'
