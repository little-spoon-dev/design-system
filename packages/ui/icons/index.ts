export { default } from '@littlespoon/icons'
export * from '@littlespoon/icons'
