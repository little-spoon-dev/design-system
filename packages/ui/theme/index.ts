export { default } from '@littlespoon/theme'
export * from '@littlespoon/theme'
