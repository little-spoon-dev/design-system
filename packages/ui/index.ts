import Alert from '@littlespoon/alert'
import Box from '@littlespoon/box'
import { Breadcrumb, Breadcrumbs } from '@littlespoon/breadcrumbs'
import Button from '@littlespoon/button'
import { Checkbox, CheckboxGroup } from '@littlespoon/checkbox'
import Divider from '@littlespoon/divider'
import Label from '@littlespoon/label'
import Link from '@littlespoon/link'
import Tag from '@littlespoon/tag'
import theme from '@littlespoon/theme'
import Typography from '@littlespoon/typography'

export {
  Alert,
  Box,
  Breadcrumb,
  Breadcrumbs,
  Button,
  Checkbox,
  CheckboxGroup,
  Divider,
  Label,
  Link,
  Tag,
  theme,
  Typography,
}

export default {
  Alert,
  Label,
  Link,
  Box,
  Breadcrumb,
  Breadcrumbs,
  Button,
  Checkbox,
  CheckboxGroup,
  Divider,
  Tag,
  theme,
  Typography,
}
