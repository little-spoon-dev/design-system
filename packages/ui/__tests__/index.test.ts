import { Breadcrumb, Breadcrumbs } from '@littlespoon/breadcrumbs'
import Button from '@littlespoon/button'
import Divider from '@littlespoon/divider'
import icons from '@littlespoon/icons'
import Tag from '@littlespoon/tag'
import theme from '@littlespoon/theme'
import Typography from '@littlespoon/typography'
import { createElement, isValidElement } from 'react'

import ui from '..'
import uiBreadcrumb from '../Breadcrumb'
import uiBreadcrumbs from '../Breadcrumbs'
import uiButton from '../Button'
import uiDivider from '../Divider'
import uiIcons from '../icons'
import uiTag from '../Tag'
import uiTheme from '../theme'
import uiTypography from '../Typography'

it('exports ui', () => {
  expect(ui).toMatchSnapshot()
})

it('exports Breadcrumb', () => {
  expect(uiBreadcrumb).toBe(Breadcrumb)
  expect(isValidElement(createElement(Breadcrumb))).toBe(true)
})

it('exports Breadcrumbs', () => {
  expect(uiBreadcrumbs).toBe(Breadcrumbs)
  expect(isValidElement(createElement(Breadcrumbs))).toBe(true)
})

it('exports Button', () => {
  expect(uiButton).toBe(Button)
  expect(isValidElement(createElement(Button))).toBe(true)
})

it('exports Divider', () => {
  expect(uiDivider).toBe(Divider)
  expect(isValidElement(createElement(Divider))).toBe(true)
})

it('exports Tag', () => {
  expect(uiTag).toBe(Tag)
  expect(isValidElement(createElement(Tag))).toBe(true)
})

it('exports Typography', () => {
  expect(uiTypography).toBe(Typography)
  expect(isValidElement(createElement(Typography))).toBe(true)
})

it('exports theme', () => {
  expect(uiTheme).toBe(theme)
})

it('exports icons', () => {
  expect(uiIcons).toBe(icons)
})
