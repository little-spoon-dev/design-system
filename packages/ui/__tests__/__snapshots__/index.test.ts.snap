// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports ui 1`] = `
{
  "Accordion": {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": [],
    "componentStyle": e {
      "baseHash": 2053189920,
      "baseStyle": undefined,
      "componentId": "sc-cyRfQX",
      "isStatic": false,
      "rules": [
        "
  width: 100%;
  display: flex;
  flex-direction: column;
",
      ],
      "staticRulesId": "",
    },
    "foldedComponentIds": [],
    "render": [Function],
    "shouldForwardProp": undefined,
    "styledComponentId": "sc-cyRfQX",
    "target": "dl",
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "Alert": [Function],
  "Box": [Function],
  "Breadcrumb": [Function],
  "Breadcrumbs": [Function],
  "Button": [Function],
  "Checkbox": [Function],
  "CheckboxGroup": [Function],
  "Divider": [Function],
  "Drawer": [Function],
  "Label": [Function],
  "Link": [Function],
  "Tag": [Function],
  "Typography": [Function],
  "theme": {
    "breakpoints": {
      "desktop": 1000,
      "down": [Function],
      "lg": 1000,
      "maxWidth": [Function],
      "md": 768,
      "minWidth": [Function],
      "mobile": 0,
      "sm": 550,
      "tablet": 768,
      "up": [Function],
      "xl": 1200,
      "xs": 375,
      "xxl": 1440,
    },
    "colors": {
      "alert": {
        "critical": {
          "critical10": [Function],
          "critical20": [Function],
          "critical30": [Function],
          "critical40": [Function],
          "critical50": [Function],
        },
        "informative": {
          "informative10": [Function],
          "informative20": [Function],
          "informative30": [Function],
          "informative40": [Function],
          "informative50": [Function],
        },
        "success": {
          "success10": [Function],
          "success20": [Function],
          "success30": [Function],
          "success40": [Function],
          "success50": [Function],
        },
        "warning": {
          "warning10": [Function],
          "warning20": [Function],
          "warning30": [Function],
          "warning40": [Function],
          "warning50": [Function],
        },
      },
      "blue10": [Function],
      "blue100": [Function],
      "blue20": [Function],
      "blue30": [Function],
      "blue40": [Function],
      "blue50": [Function],
      "blue60": [Function],
      "blue70": [Function],
      "blue80": [Function],
      "blue90": [Function],
      "borderBold": "rgba(123,123,123,1)",
      "borderContrast": "rgba(66,66,66,1)",
      "borderCritical": "rgba(237,169,159,1)",
      "borderInformative": "rgba(169,193,231,1)",
      "borderMinimal": "rgba(224,224,224,1)",
      "borderModerate": "rgba(173,173,173,1)",
      "borderSubtle": "rgba(205,205,205,1)",
      "borderSuccess": "rgba(153,199,187,1)",
      "borderWarning": "rgba(252,208,168,1)",
      "brand": "rgba(0,227,205,1)",
      "brand10": [Function],
      "brand100": [Function],
      "brand20": [Function],
      "brand30": [Function],
      "brand40": [Function],
      "brand50": [Function],
      "brand60": [Function],
      "brand70": [Function],
      "brand80": [Function],
      "brand90": [Function],
      "critical10": [Function],
      "critical20": [Function],
      "critical30": [Function],
      "critical40": [Function],
      "critical50": [Function],
      "deepGreen10": [Function],
      "deepGreen100": [Function],
      "deepGreen20": [Function],
      "deepGreen30": [Function],
      "deepGreen40": [Function],
      "deepGreen50": [Function],
      "deepGreen60": [Function],
      "deepGreen70": [Function],
      "deepGreen80": [Function],
      "deepGreen90": [Function],
      "gold10": [Function],
      "gold100": [Function],
      "gold20": [Function],
      "gold30": [Function],
      "gold40": [Function],
      "gold50": [Function],
      "gold60": [Function],
      "gold70": [Function],
      "gold80": [Function],
      "gold90": [Function],
      "grey10": [Function],
      "grey100": [Function],
      "grey20": [Function],
      "grey30": [Function],
      "grey40": [Function],
      "grey50": [Function],
      "grey60": [Function],
      "grey70": [Function],
      "grey80": [Function],
      "grey90": [Function],
      "informative10": [Function],
      "informative20": [Function],
      "informative30": [Function],
      "informative40": [Function],
      "informative50": [Function],
      "limeGreen10": [Function],
      "limeGreen100": [Function],
      "limeGreen20": [Function],
      "limeGreen30": [Function],
      "limeGreen40": [Function],
      "limeGreen50": [Function],
      "limeGreen60": [Function],
      "limeGreen70": [Function],
      "limeGreen80": [Function],
      "limeGreen90": [Function],
      "mintGreen10": [Function],
      "mintGreen100": [Function],
      "mintGreen20": [Function],
      "mintGreen30": [Function],
      "mintGreen40": [Function],
      "mintGreen50": [Function],
      "mintGreen60": [Function],
      "mintGreen70": [Function],
      "mintGreen80": [Function],
      "mintGreen90": [Function],
      "peach10": [Function],
      "peach100": [Function],
      "peach20": [Function],
      "peach30": [Function],
      "peach40": [Function],
      "peach50": [Function],
      "peach60": [Function],
      "peach70": [Function],
      "peach80": [Function],
      "peach90": [Function],
      "pink10": [Function],
      "pink100": [Function],
      "pink20": [Function],
      "pink30": [Function],
      "pink40": [Function],
      "pink50": [Function],
      "pink60": [Function],
      "pink70": [Function],
      "pink80": [Function],
      "pink90": [Function],
      "primary": {
        "brand": {
          "brand10": [Function],
          "brand100": [Function],
          "brand20": [Function],
          "brand30": [Function],
          "brand40": [Function],
          "brand50": [Function],
          "brand60": [Function],
          "brand70": [Function],
          "brand80": [Function],
          "brand90": [Function],
        },
        "primaryBlue": {
          "blue10": [Function],
          "blue100": [Function],
          "blue20": [Function],
          "blue30": [Function],
          "blue40": [Function],
          "blue50": [Function],
          "blue60": [Function],
          "blue70": [Function],
          "blue80": [Function],
          "blue90": [Function],
        },
      },
      "secondary": {
        "secondaryBlue": {
          "blue10": [Function],
          "blue100": [Function],
          "blue20": [Function],
          "blue30": [Function],
          "blue40": [Function],
          "blue50": [Function],
          "blue60": [Function],
          "blue70": [Function],
          "blue80": [Function],
          "blue90": [Function],
        },
        "secondaryDeepGreen": {
          "deepGreen10": [Function],
          "deepGreen100": [Function],
          "deepGreen20": [Function],
          "deepGreen30": [Function],
          "deepGreen40": [Function],
          "deepGreen50": [Function],
          "deepGreen60": [Function],
          "deepGreen70": [Function],
          "deepGreen80": [Function],
          "deepGreen90": [Function],
        },
        "secondaryGold": {
          "gold10": [Function],
          "gold100": [Function],
          "gold20": [Function],
          "gold30": [Function],
          "gold40": [Function],
          "gold50": [Function],
          "gold60": [Function],
          "gold70": [Function],
          "gold80": [Function],
          "gold90": [Function],
        },
        "secondaryLimeGreen": {
          "limeGreen10": [Function],
          "limeGreen100": [Function],
          "limeGreen20": [Function],
          "limeGreen30": [Function],
          "limeGreen40": [Function],
          "limeGreen50": [Function],
          "limeGreen60": [Function],
          "limeGreen70": [Function],
          "limeGreen80": [Function],
          "limeGreen90": [Function],
        },
        "secondaryMintGreen": {
          "mintGreen10": [Function],
          "mintGreen100": [Function],
          "mintGreen20": [Function],
          "mintGreen30": [Function],
          "mintGreen40": [Function],
          "mintGreen50": [Function],
          "mintGreen60": [Function],
          "mintGreen70": [Function],
          "mintGreen80": [Function],
          "mintGreen90": [Function],
        },
        "secondaryNeutralGrey": {
          "grey10": [Function],
          "grey100": [Function],
          "grey20": [Function],
          "grey30": [Function],
          "grey40": [Function],
          "grey50": [Function],
          "grey60": [Function],
          "grey70": [Function],
          "grey80": [Function],
          "grey90": [Function],
        },
        "secondaryPeach": {
          "peach10": [Function],
          "peach100": [Function],
          "peach20": [Function],
          "peach30": [Function],
          "peach40": [Function],
          "peach50": [Function],
          "peach60": [Function],
          "peach70": [Function],
          "peach80": [Function],
          "peach90": [Function],
        },
        "secondaryPink": {
          "pink10": [Function],
          "pink100": [Function],
          "pink20": [Function],
          "pink30": [Function],
          "pink40": [Function],
          "pink50": [Function],
          "pink60": [Function],
          "pink70": [Function],
          "pink80": [Function],
          "pink90": [Function],
        },
        "secondaryPurple": {
          "purple10": [Function],
          "purple100": [Function],
          "purple20": [Function],
          "purple30": [Function],
          "purple40": [Function],
          "purple50": [Function],
          "purple60": [Function],
          "purple70": [Function],
          "purple80": [Function],
          "purple90": [Function],
          "visitedLinkPurple": [Function],
        },
      },
      "shadeBlack": "#000000",
      "shadeWhite": "#FFFFFF",
      "subSurfaceCritical": "rgba(250,233,231,1)",
      "subSurfaceDull": [Function],
      "subSurfaceInformative": "rgba(233,239,249,1)",
      "subSurfaceSuccess": "rgba(230,241,238,1)",
      "subSurfaceWarning": "rgba(254,243,233,1)",
      "success10": [Function],
      "success20": [Function],
      "success30": [Function],
      "success40": [Function],
      "success50": [Function],
      "surfaceContrast": "rgba(66,66,66,1)",
      "surfaceOverlay": "rgba(66,66,66,0.8)",
      "surfacePrimary": "#FFFFFF",
      "surfaceSecondary": "rgba(241,241,241,1)",
      "surfaceStrong": "rgba(0,104,94,1)",
      "token": {
        "borderBold": "rgba(123,123,123,1)",
        "borderContrast": "rgba(66,66,66,1)",
        "borderCritical": "rgba(237,169,159,1)",
        "borderInformative": "rgba(169,193,231,1)",
        "borderMinimal": "rgba(224,224,224,1)",
        "borderModerate": "rgba(173,173,173,1)",
        "borderSubtle": "rgba(205,205,205,1)",
        "borderSuccess": "rgba(153,199,187,1)",
        "borderWarning": "rgba(252,208,168,1)",
        "brand": "rgba(0,227,205,1)",
        "shadeBlack": "#000000",
        "shadeWhite": "#FFFFFF",
        "subSurfaceCritical": "rgba(250,233,231,1)",
        "subSurfaceDull": [Function],
        "subSurfaceInformative": "rgba(233,239,249,1)",
        "subSurfaceSuccess": "rgba(230,241,238,1)",
        "subSurfaceWarning": "rgba(254,243,233,1)",
        "surfaceContrast": "rgba(66,66,66,1)",
        "surfaceOverlay": "rgba(66,66,66,0.8)",
        "surfacePrimary": "#FFFFFF",
        "surfaceSecondary": "rgba(241,241,241,1)",
        "surfaceStrong": "rgba(0,104,94,1)",
      },
      "warning10": [Function],
      "warning20": [Function],
      "warning30": [Function],
      "warning40": [Function],
      "warning50": [Function],
    },
    "fonts": {
      "primary": {
        "button": {
          "large": {
            "fontSize": "1.8rem",
            "letterSpacing": "0.15rem",
            "lineHeight": "2.6rem",
          },
          "medium": {
            "fontSize": "1.6rem",
            "letterSpacing": "0.15rem",
            "lineHeight": "2.4rem",
          },
          "small": {
            "fontSize": "1.4rem",
            "letterSpacing": "0.15rem",
            "lineHeight": "2rem",
          },
          "xlarge": {
            "fontSize": "2rem",
            "letterSpacing": "0.15rem",
            "lineHeight": "3rem",
          },
        },
        "caption": {
          "caption1": {
            "fontSize": "1.2rem",
            "lineHeight": "1.8rem",
          },
          "medium": {
            "fontSize": "1.2rem",
            "lineHeight": "1.8rem",
          },
        },
        "family": "Lato, sans-serif",
        "paragraph": {
          "large": {
            "fontSize": "1.8rem",
            "lineHeight": "2.6rem",
          },
          "medium": {
            "fontSize": "1.6rem",
            "lineHeight": "2.4rem",
          },
          "p1": {
            "fontSize": "2rem",
            "lineHeight": "3rem",
          },
          "p2": {
            "fontSize": "1.8rem",
            "lineHeight": "2.6rem",
          },
          "p3": {
            "fontSize": "1.6rem",
            "lineHeight": "2.4rem",
          },
          "p4": {
            "fontSize": "1.4rem",
            "lineHeight": "2rem",
          },
          "small": {
            "fontSize": "1.4rem",
            "lineHeight": "2rem",
          },
          "xlarge": {
            "fontSize": "2rem",
            "lineHeight": "3rem",
          },
        },
        "weight": {
          "bold": 700,
          "normal": 400,
        },
      },
      "secondary": {
        "display": {
          "display1": {
            "fontSize": "7.4rem",
            "lineHeight": "9.2rem",
          },
          "display2": {
            "fontSize": "6.6rem",
            "lineHeight": "8.2rem",
          },
        },
        "family": "Mulish, sans-serif",
        "heading": {
          "h1": {
            "fontSize": "5.2rem",
            "lineHeight": "6.5rem",
          },
          "h2": {
            "fontSize": "4.6rem",
            "lineHeight": "5.8rem",
          },
          "h3": {
            "fontSize": "3.6rem",
            "lineHeight": "4.5rem",
          },
          "h4": {
            "fontSize": "2.9rem",
            "lineHeight": "3.6rem",
          },
          "h5": {
            "fontSize": "2.6rem",
            "lineHeight": "3.2rem",
          },
          "h6": {
            "fontSize": "2rem",
            "lineHeight": "2.5rem",
          },
        },
        "weight": {
          "black": 900,
          "bold": 700,
          "extraBold": 800,
        },
      },
    },
  },
}
`;
